qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/JobWorkerTest.java

```diff
@@ -236,4 +261,118 @@ private void awaitStreamRemoved(final String jobType) {
                     .remoteStreams()
                     .doNotHaveJobType(jobType));
   }
+
+  @Nested
+  @AutoCloseResources
+  final class SlowWorkerTest {
+    private final String uniqueId = Strings.newRandomValidBpmnId();
+    private final CountDownLatch latch = new CountDownLatch(1);
+    private final BpmnModelInstance process =
+        Bpmn.createExecutableProcess(uniqueId)
+            .startEvent()
+            .serviceTask("task", b -> b.zeebeJobType(uniqueId))
+            .endEvent()
+            .done();
+
+    // Use internal gRPC constant to generate enough load to toggle the transport state faster
+    private final Map<String, Object> payload =
+        Map.of("foo", "bar".repeat(TransportState.DEFAULT_ONREADY_THRESHOLD));
+
+    private final RecordingJobHandler jobHandler = new RecordingJobHandler();
+
+    @SuppressWarnings({"FieldCanBeLocal", "unused"})
+    @AutoCloseResource
+    private JobWorker worker;
+
+    @BeforeEach
+    void beforeEach() {
+      worker =
+          client
+              .getClient()
+              .newWorker()
+              .jobType(uniqueId)
+              .handler(
+                  (c, j) -> {
+                    jobHandler.handle(c, j);
+                    latch.await();
+                  })
+              .maxJobsActive(1)
+              .pollInterval(Duration.ofMillis(10))
+              .streamEnabled(true)
+              .open();
+
+      awaitStreamRegistered(uniqueId);
+      client.deployProcess(process);
+    }
+
+    @Test
+    void shouldYieldJobIfClientIsBlocked() {
+      // given
+      // disable waiting in the exporter and rely on Awaitility polling
+      RecordingExporter.setMaximumWaitTime(0);
+
+      // when
+      Awaitility.await("until transport is suspended and jobs are yielded")
+          .untilAsserted(
+              () -> {
+                createProcessInstance(uniqueId, payload);
+
+                // then
+                assertThat(RecordingExporter.jobRecords(JobIntent.YIELDED).limit(1)).hasSize(1);
+              });
+    }
+
+    @Test
+    void shouldReceiveNewJobsWhenUnblocked() {
+      // given
```

Comment:
What was the issue in the end?

Reply:
One problem was that polling was activating the job and the list was not matching with the expected.

Then I disabled the polling, which works but I thought might be nice to have everything running together. So I enabled polling and changed the expectations.

Reply:
Good catch!